### PR TITLE
#263 Disable RegistrationWorkflow for Accounts in ITs

### PR DESCRIPTION
**Pull request by maranai**

### CODE REVIEW

File: extensions/httpclient/src/test/groovy/com/stormpath/sdk/client/DirectoryIT.groovy [hunk 1]
```diff
@@ -76,6 +76,7 @@ class DirectoryIT extends ClientIT {
             .setPassword('Changeme1!')
 
         dir.createAccount(account)
+        deleteOnTeardown(account)
```
Comment (mrioan) [inline extensions/httpclient/src/test/groovy/com/stormpath/sdk/client/DirectoryIT.groovy]: When a dir is deleted, the accounts contained inside are deleted as well. This line is not required

File: extensions/httpclient/src/test/groovy/com/stormpath/sdk/client/DirectoryIT.groovy [hunk 2]
```diff
@@ -241,6 +242,7 @@ class DirectoryIT extends ClientIT {
                 .setPassword('Changeme1!')
 
         dir.createAccount(account01)
+        deleteOnTeardown(account01)
```
Comment (mrioan) [inline extensions/httpclient/src/test/groovy/com/stormpath/sdk/client/DirectoryIT.groovy]: Idem

File: extensions/httpclient/src/test/groovy/com/stormpath/sdk/client/CustomDataEntityIT.groovy [hunk 3]
```diff
@@ -199,6 +201,8 @@ class CustomDataEntityIT extends AbstractCustomDataIT {
         //7. Get the account again and assert that the value is there:
         account = client2.getResource(account.getHref(), Account)
         assertEquals account.customData.hobby, 'Kendo'
+
+        deleteOnTeardown(account)
```
Comment (mrioan) [inline extensions/httpclient/src/test/groovy/com/stormpath/sdk/client/CustomDataEntityIT.groovy]: Not required, the account is deleted when the dir is deleted.

The `createTempApp` method does this:

    protected Application createTempApp() {
        def app = createApplication();
        deleteOnTeardown(app.getDefaultAccountStore() as Directory)
        deleteOnTeardown(app)
        return app
    }

So, the `dir` will be deleted and the `account` too.

File: extensions/httpclient/src/test/groovy/com/stormpath/sdk/client/CustomDataEntityIT.groovy [hunk 4]
```diff
@@ -49,6 +49,8 @@ class CustomDataEntityIT extends AbstractCustomDataIT {
 
         deleteOnTeardown(account.directory)
 
+        deleteOnTeardown(account)
```
Comment (mrioan) [inline extensions/httpclient/src/test/groovy/com/stormpath/sdk/client/CustomDataEntityIT.groovy]: Not required, the account is deleted when the dir is deleted.

File: extensions/httpclient/src/test/groovy/com/stormpath/sdk/impl/application/ApplicationIT.groovy [hunk 5]
```diff
@@ -106,6 +107,7 @@ class ApplicationIT extends ClientIT {
         account.password = 'Changeme1!'
 
         def created = app.createAccount(account)
+        deleteOnTeardown(account)
```
Comment (mrioan) [inline extensions/httpclient/src/test/groovy/com/stormpath/sdk/impl/application/ApplicationIT.groovy]: Not needed

File: extensions/httpclient/src/test/groovy/com/stormpath/sdk/impl/resource/CollectionIterationIT.groovy [hunk 6]
```diff
@@ -88,6 +88,7 @@ class CollectionIterationIT extends ClientIT {
                     .setEmail(uniquify("john${i}deleteme") + "@stormpath.com")
                     .setPassword('Changeme1!')
             dir.createAccount(account);
+            deleteOnTeardown(account)
```
Comment (mrioan) [inline extensions/httpclient/src/test/groovy/com/stormpath/sdk/impl/resource/CollectionIterationIT.groovy]: not needed

**Comment by maranai** [comment]
@mrioan I've removed the unnecessary deleteOnTearDown. Ready for a new review round. Thanks!

**Comment by mrioan** [comment]
@maranai it looks good. Approved. You can merge. Thanks!

**Pull request merged**